### PR TITLE
Metric collection for Dart ecosystem

### PR DESCRIPTION
**Pull request by sachin-sandhu**
### What are you trying to accomplish?

Adds metric collection for Dart language ecosystem along with pub package manager

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: pub/lib/dependabot/pub/package_manager.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      VERSION = "0.0"
+
+      SUPPORTED_VERSIONS = T.let([].freeze, T::Array[Dependabot::Version])
+
+      DEPRECATED_VERSIONS = T.let([].freeze, T::Array[Dependabot::Version])
+
+      sig do
+        params(
+          raw_version: String,
+          requirement: T.nilable(Requirement)
```
Comment (amazimbe) [inline pub/lib/dependabot/pub/package_manager.rb:30]: I think you should simplify by removing this require from here and below.

File: pub/lib/dependabot/pub/language.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+module Dependabot
+  module Pub
+    LANGUAGE = "dart"
+
+    class Language < Dependabot::Ecosystem::VersionManager
+      extend T::Sig
+
+      sig { params(raw_version: String, requirement: T.nilable(Requirement)).void }
+      def initialize(raw_version, requirement = nil)
```
Comment (amazimbe) [inline pub/lib/dependabot/pub/language.rb:16]: same here

File: pub/lib/dependabot/pub/package_manager.rb (hunk 1 above)
Comment (sachin-sandhu) [reply pub/lib/dependabot/pub/package_manager.rb]: @amazimbe , fixed.

File: pub/lib/dependabot/pub/language.rb (hunk 2 above)
Comment (sachin-sandhu) [reply pub/lib/dependabot/pub/language.rb]: @amazimbe , fixed.

**Review by amazimbe — APPROVED** [review]

File: pub/lib/dependabot/pub/package_manager.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# typed: strong
```
Comment (markhallen) [inline pub/lib/dependabot/pub/package_manager.rb:1]: TIL; `typed: strong` - [Banning untyped from a file · Sorbet](https://sorbet.org/docs/strong).

👍

**Pull request merged**